Mage/src/main/java/mage/game/events/CreateTokenEvent.java

```diff
@@ -19,9 +20,14 @@ public class CreateTokenEvent extends GameEvent {
      * @param amount
      * @param token
      */
-    public CreateTokenEvent(Ability source, UUID controllerId, int amount, Token token) {
+    public CreateTokenEvent(Ability source, UUID controllerId, int amount, Token token, List<Token> additionalTokens) {
```

Comment:
Use one list for all tokens. No needs in main + additional list. For text gen, for put, etc.

Reply:
The reason there's a main and an additional list is because that's how tokens are created, via the `token.putOntoBattlefield` in `CreateTokenEffect` - that's why there's a main token (the one that's put into `CreateTokenEffect`) and additional tokens (the other ones to be added when constructing `CreateTokenEvent`); doing it with one list would require decomposing it into a main + additional list within the `CreateTokenEffect`.

Reply:
I don't see the problem with a separate list for the additional tokens (it's rare usage relative to all CreateTokenEffect)

Reply:
mainToken = tokensList.get(0)

Reply:
Alright, redid it to have a single list, and now `token.putOntoBattlefield()` technically ignores itself in favor of an additional `tokens` parameter for `putOntoBattlefield` (but for existing usage of `token.putOntoBattlefield()` outside of `CreateTokenEffect` it will auto-populate the `tokens` parameter with `Arrays.asList(this)`)